### PR TITLE
Drop standby tasks in standby taks executor 

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
1. Revert "Drop task when namespace is not on the cluster (#4444)"
2. Add logic to drop standby task in transfer / timer standby task executor.

**Why?**

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

**Review by yycptt — APPROVED** [review]

File: service/history/queues/executor_wrapper_test.go
Comment (yycptt) [inline service/history/queues/executor_wrapper_test.go:1]: why remove this?

File: service/history/timer_queue_standby_task_executor.go [hunk 1]
```diff
@@ -459,6 +459,11 @@ func (t *timerQueueStandbyTaskExecutor) processTimer(
 		// workflow already finished, no need to process the timer
 		return nil
 	}
+	nsRecord := mutableState.GetNamespaceEntry()
+	if !nsRecord.IsOnCluster(t.clusterName) {
+		// discard standby tasks
+		return nil
+	}
```
Comment (yycptt) [inline service/history/timer_queue_standby_task_executor.go:466]: nit: I still think this logic should live in `getStandbyPostActionFn`. `processTimer/Transfer` can be used for any task logic, so any new task might get dropped accidentally. But `getStandbyPostActionFn` is specifically for deciding what to do if a standby task can't be processed, and when invoking this function, developer is aware of the fact that the discard function he/she provided will be used. 

Although I understanding changes here will prevent task from being executed at all. So I don't have strong opinion.

**Review by yycptt — COMMENTED** [review]
Plz update the PR title as well.

File: service/history/timer_queue_standby_task_executor.go (hunk 1 above)
Comment (yux0) [reply service/history/timer_queue_standby_task_executor.go]: Yes. I try to prevent task being executed compare execute once and discard.

**Review by yycptt — APPROVED** [review]

**Pull request merged**